### PR TITLE
Remove NamespaceSourceQuerier API that is no longer in use

### PR DESCRIPTION
**Pull request by dinhxuanvu**
NamespaceSourceQuerier API is no longer in use for new resolver so it
should be removed to reduce tech debt and unusable code.

Signed-off-by: Vu Dinh <vudinh@outlook.com>

**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/controller/operators/catalogtemplate/operator.go [hunk 1]
```diff
@@ -130,6 +130,11 @@ func (o *Operator) syncCatalogSources(obj interface{}) error {
 		return nil
 	}
 
+	// This catalog template feature is not usable in configmap/internal catalogsource
+	if inputCatalogSource.Spec.SourceType == v1alpha1.SourceTypeInternal || inputCatalogSource.Spec.SourceType == v1alpha1.SourceTypeConfigmap {
+		return nil
+	}
```
Comment (timflannagan) [inline pkg/controller/operators/catalogtemplate/operator.go:136]: How do you feel about splitting these changes into their own PR - namely in the context of record keeping downstream?

Comment (dinhxuanvu) [reply pkg/controller/operators/catalogtemplate/operator.go:136]: I'm fine with that. Will do so after I see the result from CI e2e-tests. I want to take this opportunity to see how it is doing.

**Comment by timflannagan** [comment]
@dinhxuanvu Can you rebase?

**Pull request merged**